docs/README.md

```diff
@@ -1,11 +1,27 @@
-- [Troubleshooting build errors](BuildErrors.md)
-- [Artifacts structure](Artifacts.md)
+The table below outlines the different docs in this folder and what they are helpful for.
+
+| Documentation                                                            | What is it about?                                                       | Who is it for?                                                                                                              |
+|--------------------------------------------------------------------------|-------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------|
+| [API review process](APIReviewProcess.md)                                | Outlines the process for reviewing API changes in ASP.NET Core          | Anyone looking to understand the process for making API changes to ASP.NET Core                                             |
+| [Artifacts structure](Artifacts.md)                                      | Outlines the artifacts produced by the build                            | Anyone looking to understand artifiacts produced from an Azure DevOps build                                                 |
+| [Troubleshooting build errors](BuildErrors.md)                           | Common errors that occur when building the repo and how to resolve them | Anyone running into an issue with the build locally                                                                         |
+| [Building from source](BuildFromSource.md)                               | Setup instructions for the ASP.NET Core repo                            | First-time contributors                                                                                                     |
+| [Working with EventSources and EventCounters](EventSourceAndCounters.md) | Guidance on adding event tracing to a library                           | Contributors needing to add event tracing for diagnostics purposes                                                          |
+| [Tests on Helix](Helix.md)                                               | An overview of the Helix test environment                               | Contributors debugging tests in Helix or looking to understand the output from Helix builds                                 |
+| [Issue management](IssueManagementPolicies.md)                           | Overview of policies in place to manage issues                          | Community members and collaborators looking to understand how we handle closed issue, issues that need author feedback, etc |
+| [Package archives](PackageArchives.md)                                   |                                                                         |                                                                                                                             |
```

Comment:
I don't quite grok what to fill in for this document.

@dougbu @wtgodbe Maybe you know?